controllers/activemqartemis_controller_test.go

```diff
@@ -122,6 +122,236 @@ var _ = Describe("artemis controller", func() {
+				g.Expect(len(credSecret.OwnerReferences) > 0).Should(BeTrue())
+				ownerFound := false
+				for _, oref := range credSecret.OwnerReferences {
+					if oref.Kind == "ActiveMQArtemis" && oref.Name == brokerCr.Name {
+						ownerFound = true
+						break
+					}
+				}
+				g.Expect(ownerFound).To(BeTrue())
+
+				g.Expect(k8sClient.Get(ctx, ssKey, createdSs)).Should(Succeed())
+
+				initContainer := createdSs.Spec.Template.Spec.InitContainers[0]
+				userFound := false
+				passwordFound := false
+				clusterUserFound := false
+				clusterPasswordFound := false
+				for _, envVar := range initContainer.Env {
+					if envVar.Name == "AMQ_USER" {
+						userFound = true
+						g.Expect(envVar.ValueFrom.SecretKeyRef.Name).To(Equal(brokerCr.Name+"-credentials-secret"), envVar)
+						g.Expect(envVar.ValueFrom.SecretKeyRef.Key).To(Equal("AMQ_USER"), envVar)
+					}
+					if envVar.Name == "AMQ_PASSWORD" {
+						passwordFound = true
+						g.Expect(envVar.ValueFrom.SecretKeyRef.Name).To(Equal(brokerCr.Name+"-credentials-secret"), envVar)
+						g.Expect(envVar.ValueFrom.SecretKeyRef.Key).To(Equal("AMQ_PASSWORD"), envVar)
+					}
+					if envVar.Name == "AMQ_CLUSTER_USER" {
+						clusterUserFound = true
+						g.Expect(envVar.ValueFrom.SecretKeyRef.Name).To(Equal(brokerCr.Name+"-credentials-secret"), envVar)
+						g.Expect(envVar.ValueFrom.SecretKeyRef.Key).To(Equal("AMQ_CLUSTER_USER"), envVar)
+					}
+					if envVar.Name == "AMQ_CLUSTER_PASSWORD" {
+						clusterPasswordFound = true
+						g.Expect(envVar.ValueFrom.SecretKeyRef.Name).To(Equal(brokerCr.Name+"-credentials-secret"), envVar)
+						g.Expect(envVar.ValueFrom.SecretKeyRef.Key).To(Equal("AMQ_CLUSTER_PASSWORD"), envVar)
+					}
+				}
+				g.Expect(userFound).To(BeTrue())
+				g.Expect(passwordFound).To(BeTrue())
+				g.Expect(clusterUserFound).To(BeTrue())
+				g.Expect(clusterPasswordFound).To(BeTrue())
+
+			}, timeout, interval).Should(Succeed())
+
+			// cleanup
+			Expect(k8sClient.Delete(ctx, brokerCr)).Should(Succeed())
+			By("check it has gone")
+			Eventually(func() bool {
+				return checkCrdDeleted(brokerCr.Name, defaultNamespace, createdBrokerCr)
+			}, timeout, interval).Should(BeTrue())
+		})
+
+		It("user credential secret", func() {
+			brokerCrName := "broker-user-cred"
+			userSecretName := brokerCrName + "-credentials-secret"
+			secretData := make(map[string]string)
+			secretData["AMQ_USER"] = "myuser"
+			secretData["AMQ_PASSWORD"] = "mypassword"
+			secretData["AMQ_CLUSTER_USER"] = "myclusteruser"
+			secretData["AMQ_CLUSTER_PASSWORD"] = "myclusterpassword"
+
+			By("deploy a user credential secret")
+			secret, createdSecret := DeploySecret(defaultNamespace, func(candidate *corev1.Secret) {
+				candidate.Name = userSecretName
+				candidate.StringData = secretData
+			})
+
+			By("deploy a broker")
+			brokerCr, createdBrokerCr := DeployCustomBroker(defaultNamespace, func(candidate *brokerv1beta1.ActiveMQArtemis) {
+				candidate.Name = brokerCrName
```

Comment:
Could we set custom `adminUser` and `adminPassword` to check their values in the credential secret updated by the operator?
I guess the test should fail because the secret isn't owned by the operator but it used to work in previous versions.

Reply:
yes I'll add one test. But yes we would expect the secret not to get updated. (I'll test it as such as it is current behavior). We can update it if we decide to do otherwise.